services/template.py

```diff
@@ -0,0 +1,19 @@
+from jinja2 import Environment, PackageLoader, StrictUndefined, select_autoescape
+
+
+def get_template_service():
+    return TemplateService()
+
+
+class TemplateService:
+    def __init__(self):
+        # this loads the templates from the templates directory in this repository since it's looking for a dir named `templates` next to app.py
+        self.env = Environment(
+            loader=PackageLoader("app"),
+            autoescape=select_autoescape(),
+            undefined=StrictUndefined,
+        )
+
+    def get_template(self, name, **kwargs):
```

Comment:
[nit] change "name" to "template_name".

[less nit]
Personally I don't love that we return a rendered template from this function. I think that this function should simply get and return the template we specified, AND handle errors when we fail to get the template (like you ask for one that doesn't exist)

Then we modify the template object as we see fit and render it ourselves later on.

So I do think the responsibilities of this function should be limited to fetching a template and returning it (as the name suggests) and handling errors.